app/lib/actions/katello/applicability/repository/regenerate.rb

```diff
@@ -11,9 +11,14 @@ class Regenerate < Actions::EntryAction
           end
 
           def run
-            ::Katello::Repository.find(input[:repo_id]).content_facets.each do |facet|
-              ::Katello::EventQueue.push_event(::Katello::Events::GenerateHostApplicability::EVENT_TYPE, facet.host.id)
+            hosts = ::Katello::Repository.find(input[:repo_id]).hosts_with_applicability
```

Comment:
This is kind of a nitpick, but since there's a test failure (that might be an ordering issue), might as well update this too: 
This could return 10s of thousands of hosts, so i'd probably optimize here to only return ids:
```host_ids = ::Katello::Repository.find(input[:repo_id]).hosts_with_applicability.pluck(:id)```
and not require the full host to be loaded